src/Magnum/Math/Geometry/Test/IntersectionTest.cpp

```diff
@@ -124,20 +144,192 @@ void IntersectionTest::pointFrustum() {
+        {0.0f, -1.0f, 0.0f, 1.0f},
         {0.0f, 0.0f, 1.0f, 0.0f},
         {0.0f, 0.0f, -1.0f, 10.0f}};
 
-    CORRADE_VERIFY(Intersection::boxFrustum({Vector3{1.0f}, Vector3{2.0f}}, frustum));
+    /* Fully inside */
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{0.0f}, Vector3{1.0f}, frustum));
+    /* Intersects with exactly one plane each */
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{2.5f, 0.0f, 5.0f}, Vector3{0.1f}, frustum));
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{2.5f, 1.0f, 5.0f}, Vector3{0.1f}, frustum));
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{0.0f, 0.5f, 5.0f}, Vector3{0.1f}, frustum));
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{5.0f, 0.5f, 5.0f}, Vector3{0.1f}, frustum));
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{2.5f, 0.5f, 0.0f}, Vector3{0.1f}, frustum));
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{2.5f, 0.5f, 10.0f}, Vector3{0.1f}, frustum));
     /* Bigger than frustum, but still intersects */
-    CORRADE_VERIFY(Intersection::boxFrustum(Range3D{Vector3{-100.0f}, Vector3{100.0f}}, frustum));
+    CORRADE_VERIFY(Intersection::aabbFrustum(Vector3{0.0f}, Vector3{100.0f}, frustum));
     /* Outside of frustum */
-    CORRADE_VERIFY(!Intersection::boxFrustum(Range3D{Vector3{-10.0f}, Vector3{-5.0f}}, frustum));
+    CORRADE_VERIFY(!Intersection::aabbFrustum(Vector3{-7.5f}, Vector3{2.5f}, frustum));
+}
+
+void IntersectionTest::sphereFrustum() {
+    const Frustum frustum{
+        {1.0f, 0.0f, 0.0f, 0.0f},
+        {-1.0f, 0.0f, 0.0f, 10.0f},
+        {0.0f, 1.0f, 0.0f, 0.0f},
+        {0.0f, -1.0f, 0.0f, 10.0f},
+        {0.0f, 0.0f, 1.0f, 0.0f},
+        {0.0f, 0.0f, -1.0f, 10.0f}};
+
+    /* Sphere on edge */
+    CORRADE_VERIFY(Intersection::sphereFrustum({0.0f, 0.0f, -1.0f}, 1.5f, frustum));
+    /* Sphere inside */
+    CORRADE_VERIFY(Intersection::sphereFrustum({5.5f, 5.5f, 5.5f}, 1.5f,  frustum));
+    /* Sphere outside */
+    CORRADE_VERIFY(!Intersection::sphereFrustum({0.0f, 0.0f, 100.0f}, 0.5f, frustum));
+}
+
+void IntersectionTest::pointCone() {
+    const Vector3 center{0.1f, 0.2f, 0.3f};
+    const Vector3 normal{Vector3{0.5f, 1.0f, 2.0f}.normalized()};
+    const Rad angle{Deg{72.0f}};
+
+    /* Some vector along the surface of the cone */
+    auto surface = Matrix4::rotation(0.5f*angle, Vector3::yAxis()).transformVector(normal);
+    /* Tangent to that surface vector */
+    auto tangent = Matrix4::rotation(Deg{90.0f}, Vector3::yAxis()).transformVector(surface);
+
+    /* Point on edge */
+    CORRADE_VERIFY(Intersection::pointCone(center, center, normal, angle));
+    /* Point inside */
+    CORRADE_VERIFY(Intersection::pointCone(center + normal, center, normal, angle));
+    /* Point outside */
+    CORRADE_VERIFY(!Intersection::pointCone(Vector3{}, center, normal, angle));
+    //TODO: Why does this fail? CORRADE_VERIFY(!Intersection::pointCone(center + 5.0f*surface + 0.1f*tangent, center, normal, angle));
```

Comment:
Is this TODO still relevant?

Reply:
Yes, I should have a look at that, (sorry)

Reply:
![image](https://user-images.githubusercontent.com/1889284/38379673-b61d513e-3901-11e8-8ac5-c536ac6c56a3.png)
Well, not outside after all. Should be, though, wanted it to "just" not touch the surface.